AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_entity_utils.py

```diff
@@ -58,7 +74,128 @@ def get_property_tree(self):
+        """
+        assert (
+            self.is_property_container(component_property_path)
+        ), f"Failure: '{component_property_path}' is not a property container"
+        container_count_outcome = self.property_tree.get_container_count(component_property_path)
+        assert (
+            container_count_outcome.IsSuccess()
+        ), f"Failure: get_container_count did not return success for '{component_property_path}'"
+        return container_count_outcome.GetValue()
+
+    def reset_container(self, component_property_path: str):
+        """
+        Used to rest a container to empty
+        :param component_property_path: String of component property. (e.g. 'Settings|Visible')
+        :return: None
+        """
+        assert (
+            self.is_property_container(component_property_path)
+        ), f"Failure: '{component_property_path}' is not a property container"
+        reset_outcome = self.property_tree.reset_container(component_property_path)
+        assert (
+            reset_outcome.IsSuccess()
+        ), f"Failure: could not reset_container on '{component_property_path}'"
+
+    def append_container_item(self, component_property_path: str, value: any):
+        """
+        Used to append a container item without providing an index key.
+        :param component_property_path: String of component property. (e.g. 'Settings|Visible')
+        :param value: Value to be set
+        :return: None
+        """
+        assert (
+            self.is_property_container(component_property_path)
+        ), f"Failure: '{component_property_path}' is not a property container"
+        append_outcome = self.property_tree.append_container_item(component_property_path, value)
+        assert (
+            append_outcome.IsSuccess()
+        ), f"Failure: could not append_container_item to '{component_property_path}'"
+
+    def add_container_item(self, component_property_path: str, key: any, value: any):
+        """
+        Used to add a container item at a specified key. In practice key should be an integer index.
+        :param component_property_path: String of component property. (e.g. 'Settings|Visible')
+        :param key: Zero index integer key, although this could be any unique unused key value
+        :param value: Value to be set
+        :return: None
+        """
+        assert (
+            self.is_property_container(component_property_path)
+        ), f"Failure: '{component_property_path}' is not a property container"
+        add_outcome = self.property_tree.add_container_item(component_property_path, key, value)
+        assert (
+            add_outcome.IsSuccess()
+        ), f"Failure: could not add_container_item '{key}' to '{component_property_path}'"
+
+    def get_container_item(self, component_property_path: str, key: any) -> any:
+        """
+        Used to retrieve a container item value at the specified key. In practice key should be an integer index.
```

Comment:
the get_container_count can be used to determine if key of an indexed container is less than count. in the case of an associative container which behaves like a dictionary in some ways a non-existent key will result in an outcome.IsSuccess() False I will investigate if it is possible to get more information from outcome.GetError(). possibly something can be written for has_key using the pattern of get_container_item but returning only the IsSuccess(). No specific property tree editor method exists for key exists.

Reply:
added get_outcome.GetError() to the get_container_item failure assert. a limited number of message come out of the PTE so this may be of some value but hard to say.
attempting to access a key that does not exist falls through to "GetContainerItem - keyed value could not be returned"
non-numeric key for indexed container returns "GetContainerItem - key value type should convert into size_t"
an associative container bad key would likely return "GetContainerItem - invalid key pair for associative container"